test/integration/dra/dra_test.go

```diff
@@ -700,36 +700,56 @@ func testPrioritizedList(tCtx ktesting.TContext, enabled bool) {
+			Expression: fmt.Sprintf("device.driver == %q", nonExistentDriver),
+		},
+	}}
+
+	c, err := tCtx.Client().ResourceV1().DeviceClasses().Create(tCtx, testClass, metav1.CreateOptions{})
 	tCtx.ExpectNoError(err, "create class")
+	tCtx.CleanupCtx(func(tCtx ktesting.TContext) {
+		tCtx.Log("Cleaning up DeviceClass...")
+		err := tCtx.Client().ResourceV1().DeviceClasses().Delete(tCtx, c.Name, metav1.DeleteOptions{})
+		tCtx.ExpectNoError(err, "delete DeviceClass")
+	})
 	if enabled {
 		require.NotEmpty(tCtx, c.Spec.ExtendedResourceName, "should store ExtendedResourceName")
 	}
-	namespace := createTestNamespace(tCtx, nil)
 	tCtx.Run("scheduler", func(tCtx ktesting.TContext) {
 		startScheduler(tCtx)
 
 		pod := podWithExtendedResource.DeepCopy()
 		pod.Namespace = namespace
 		_, err := tCtx.Client().CoreV1().Pods(namespace).Create(tCtx, pod, metav1.CreateOptions{})
 		tCtx.ExpectNoError(err, "create pod")
+		// When DRAExtendedResource feature is enabled, the extended resource is delegated to DRA
+		// (because there's a DeviceClass with ExtendedResourceName set), so DRA handles the scheduling
+		// and reports "cannot allocate all claims" when no devices are available.
+		// When the feature is disabled, the noderesources plugin handles it and reports "Insufficient".
+		// Since this pod has no ResourceClaims, DRA's PostFilter returns different messages:
+		// - When DRA handles the extended resource (feature enabled): "still not schedulable" (tried but failed)
+		// - When noderesources handles it (feature disabled): "no new claims to deallocate" (no claims to process)
+		var expectedMessage string
+		if enabled {
+			expectedMessage = "0/2 nodes are available: 2 cannot allocate all claims. still not schedulable, preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling."
+		} else {
+			expectedMessage = "0/2 nodes are available: 2 Insufficient my-example.com/my-extended-resource. no new claims to deallocate, preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling."
+		}
 		schedulingAttempted := gomega.HaveField("Status.Conditions", gomega.ContainElement(
 			gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
 				"Type":    gomega.Equal(v1.PodScheduled),
 				"Status":  gomega.Equal(v1.ConditionFalse),
 				"Reason":  gomega.Equal("Unschedulable"),
-				"Message": gomega.Equal("0/2 nodes are available: 2 Insufficient my-example.com/my-extended-resource. no new claims to deallocate, preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling."),
+				"Message": gomega.Equal(expectedMessage),
 			}),
 		))
-		if enabled {
-			// pod can be scheduled as the drivers in testPublishResourceSlices provide the devices.
-			schedulingAttempted = gomega.HaveField("Status.Conditions", gomega.ContainElement(
-				gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
-					"Type":   gomega.Equal(v1.PodScheduled),
-					"Status": gomega.Equal(v1.ConditionTrue),
-				}),
-			))
```

Comment:
it was able to be scheduled before when enabled is true, why it cannot be scheduled anymore?